### PR TITLE
[jit] Fix parsing of big float literals

### PR DESCRIPTION
**Pull request by driazati**
Differential Revision: [D19186604](https://our.internmc.facebook.com/intern/diff/19186604/)

### CODE REVIEW

File: torch/csrc/jit/script/tree_views.h [hunk 1]
```diff
@@ -794,14 +794,15 @@ struct Const : public Expr {
     tree_->matchNumSubtrees(TK_CONST, 1);
   }
   bool isFloatingPoint() const {
-    return subtree(0)->stringValue().find_first_of(".eE") != std::string::npos;
```
Comment (resistor) [inline torch/csrc/jit/script/tree_views.h]: Why not just call `strtod` here, similar to how you do for `asIntegral`?

Comment (driazati) [reply torch/csrc/jit/script/tree_views.h]: If `strtod` can’t do the convention it returns 0, so it can’t determine whether a string is a valid float

Comment (resistor) [reply torch/csrc/jit/script/tree_views.h]: You can verify whether it succeeded in the conversion by inspecting `endptr`.

Comment (eellison) [reply torch/csrc/jit/script/tree_views.h]: @driazati is there a reason not to do the suggestion here ?

Comment (driazati) [reply torch/csrc/jit/script/tree_views.h]: `strtod` succeeds for ints and floats (e.g. `strtod("12")` returns `12.0`)

**Pull request closed without merging**